optimizely/odp/odp_event_manager.py

```diff
@@ -0,0 +1,187 @@
+                    self.event_queue.task_done()
+                    continue
+
+                if isinstance(item, OdpEvent):
+                    self._add_to_batch(item)
+                    self.event_queue.task_done()
+
+        except Exception as exception:
+            self.logger.error(f'Uncaught exception processing ODP events. Error: {exception}')
+
+        finally:
+            self.logger.info('Exiting ODP event processing loop. Attempting to flush pending events.')
+            self._flush_batch()
+
+    def flush(self) -> None:
+        """ Adds flush signal to event_queue. """
+
+        self.event_queue.put(self._FLUSH_SIGNAL)
+
+    def _flush_batch(self) -> None:
+        """ Flushes current batch by dispatching event. """
+        batch_len = len(self._current_batch)
+        if batch_len == 0:
+            self.logger.debug('Nothing to flush.')
+            return
+
+        api_key = self.odp_config.get_api_key()
+        api_host = self.odp_config.get_api_host()
+
+        if not api_key or not api_host:
+            self.logger.debug('ODP event processing has been disabled.')
+            return
+
+        self.logger.debug(f'Flushing batch size {batch_len}.')
+        should_retry = False
+        with self.lock:
+            event_batch = list(self._current_batch)
+            try:
+                should_retry = self.zaius_manager.send_odp_events(api_key, api_host, event_batch)
+            except Exception as e:
+                self.logger.error(Errors.ODP_EVENT_FAILED.format(f'{event_batch} {e}'))
+
+            if should_retry:
+                self.logger.debug('Error dispatching ODP events, scheduled to retry.')
+                return
+
+            self._current_batch = []
+
+    def _add_to_batch(self, odp_event: OdpEvent) -> None:
+        """ Method to append received odp event to current batch."""
+
+        with self.lock:
+            self._current_batch.append(odp_event)
+        if len(self._current_batch) >= self.batch_size:
+            self.logger.debug('Flushing ODP events on batch size.')
+            self._flush_batch()
+
+    def stop(self) -> None:
+        """ Stops and disposes batch odp event queue."""
+        self.event_queue.put(self._SHUTDOWN_SIGNAL)
+        self.logger.warning('Stopping ODP Event Queue.')
+
+        if self.is_running:
+            self.executor.join()
+
+        if len(self._current_batch) > 0:
+            self.logger.error(Errors.ODP_EVENT_FAILED.format(self._current_batch))
+
+        if self.is_running:
+            self.logger.error('Error stopping ODP event queue.')
+
+    def send_event(self, type: str, action: str, identifiers: dict[str, str], data: dict[str, Any]) -> None:
+        event = OdpEvent(type, action, identifiers, data)
+        self.dispatch(event)
+
+    def dispatch(self, event: OdpEvent) -> None:
+        if not self.odp_config.odp_integrated():
+            self.logger.debug('ODP event processing has been disabled.')
+            return
```

Comment:
I was thinking we would just block on datafile ready when calling `OptimizelyUserContext.fetch_qualified_segments`, so that becomes a non issue. Especially since on server, Optimizely client initialization should(?) be an infrequent occurrence.

If you think that's not a good solution, I can set it up similarly to how you have it implemented in swift?